gtt/gtt_display.py

```diff
@@ -182,3 +182,417 @@ def update_bar_value(self, bar_id: IdType, value: int):
+        """
+
+        self._validate_x(x_pos, x_pos + width - 1)
+        self._validate_y(y_pos, y_pos + height - 1)
+        region_id = self._resolve_id(region_id, new=True)
+
+        self._conn.write(
+            bytes.fromhex('FE  96') +
+            region_id.to_bytes(1, 'big') +
+            ints_to_signed_shorts(x_pos, y_pos) +
+            ints_to_unsigned_shorts(width, height) +
+            off_id.to_bytes(1, 'big') +
+            on_id.to_bytes(1, 'big')
+        )
+
+        self._receive_status_response(252, 150)
+
+    def set_GPO_state(self, pin: int, state: bool):
+        """Toggle the specified General Purpose Output pin on or off, sourcing up to 15mA current at
+        5V per GPO or sinking to ground. This command can be used to control devices, or signal a host device.
+        :param pin: GPO to be controlled
+        :param state: GPO state, as per eGPOSetting Values
+        """
+
+        self._conn.write(
+            bytes.fromhex('FE  49') +
+            pin.to_bytes(1, 'big') +
+            state.to_bytes(1, 'big')
+        )
+
+    def setup_animation(self, animation_id: IdType, animation_instance: int, x_pos: int, y_pos: int, fileName: str,
+                        play=True):
+        """Define a region of the screen to be used for the specified animation. If an animation is already in use
+        at that index, it will be overwritten. Multiple Animation Instances can be setup from one buffered animation
+        file.
+        :param animation_id: index where an animation file has been loaded.
+        :param animation_instance: index used to identify this animation instance in the animation list.
+        :param x_pos: the distance from the left edge of the screen in pixels
+        :param y_pos: the distance from the top edge of the screen in pixels
+        :param fileName: filename, and path from the root folder, of the animation file to load.
+        :param play: desired animation state
+        """
+        animation_id = self._resolve_id(animation_id, new=True)
+        self._conn.write(
+            bytes.fromhex('FE  C1') +
+            animation_id.to_bytes(1, 'big') +
+            animation_instance.to_bytes(1, 'big') +
+            ints_to_signed_shorts(x_pos, y_pos)
+        )
+
+        self.load_animation(animation_id, fileName, animation_instance, play)
+
+    def load_animation(self, animation_id: IdType, fileName: str, animation_instance: int,
+                       play=True):
+        """Loads an animation file from the SD card into an animation buffer for use.
+        :param animation_id: index used to identify this animation file. Specific to animations.
+        :param fileName: filename, and path from the root folder, of the animation file to load.
+        :param animation_instance: index used to identify this animation instance in the animation list.
+        :param play: desired animation state
+        """
```

Comment:
- This function should not take `animation_instance` and `play` as parameters
- snake_case `fileName`
- give a tutorial of how to create an animation file in the docstring
- note that this can take a little while to complete